src/openloco/window.cpp

```diff
@@ -387,16 +387,19 @@ namespace openloco::ui
 
     // 0x00459E54
     // TODO: needs expansion in terms of (output) parameters.
-    static void get_map_coordinates_from_pos(int32_t screenX, int32_t screenY, int32_t flags, int16_t* x, int16_t* y)
+    viewport_interaction::InteractionItem get_map_coordinates_from_pos(int32_t screenX, int32_t screenY, int32_t flags, int16_t* x, int16_t* y, void** dest)
     {
         registers regs;
         regs.ax = screenX;
-        regs.cx = screenY;
+        regs.bx = screenY;
         regs.edx = flags;
         call(0x00459E54, regs);
 
         *x = regs.ax;
         *y = regs.cx;
```

Comment:
Does this line need changing too, or is this perhaps where my original confusion (b2e1f298) stems from?

Reply:
I think Sawyer uses different default registers for different coordinate systems. The 3D coordinates tend to go in `ax`, `cx`, and occasionally `dx` for height, whereas the 2D screen-space coordinates tend to go in `ax` and `bx`. That said, I haven't looked into this function enough to give a solid answer.

Reply:
Good to know! Thanks, I'll keep this in mind.